mat64/shadow.go

```diff
@@ -191,15 +191,44 @@ func (v *Vector) checkOverlap(a blas64.Vector) bool {
 // rectanglesOverlap returns whether the strided rectangles a and b overlap
 // when b is offset by off elements after a but has at least one element before
 // the end of a. a and b have aCols and bCols respectively.
+//
+// rectanglesOverlap works by shifting both matrices left such that the left
+// column of a is at 0. The column indexes are flattened by obtaining the shifted
+// relative left and right column positions modulo the common stride. This allows
+// direct comparison of the column offsets when the matrix backing data slices
+// are known to overlap.
 func rectanglesOverlap(off, aCols, bCols, stride int) bool {
 	if stride == 1 {
+		// Unit stride means overlapping data
+		// slices must overlap as matrices.
 		return true
 	}
+
+	// Flatten the shifted matrix column positions
+	// so a starts at 0, modulo the common stride.
+	const aFrom = 0
 	aTo := aCols
+	// The mod stride operations here make the from
+	// and to indexes comparable between a and b when
+	// the data slices of a and b overlap.
 	bFrom := off % stride
 	bTo := (bFrom + bCols) % stride
+
+	// Compare the case where the b matrix is
+	// not wrapped.
 	if bFrom < bTo {
 		return aTo > bFrom
 	}
-	return bTo > 0
+
+	// Compare the case where the right column
+	// of b aligns with the stride of the backing
+	// data slice.
+	if bTo == 0 {
+		return bFrom < aTo
+	}
+
+	// Compare the case where the right column of
+	// b wraps around to the left again, modulo
+	// stride.
+	return bTo > aFrom
```

Comment:
This is the same as `return true`, right? It's clearer to me to read something like "b wraps, so must overlap".